provider/src/requestProcessor.ts

```diff
@@ -50,6 +51,34 @@ export const processSingleRequest = async (
     }
   }
 
+  // Check if this is an ENS/Basename request that can be cached
+  if (isNameResolutionRequest(request, networkId)) {
+    // Try to get from cache
+    const cachedData = await getENSResponseFromKV(request, networkId, env)
+    if (cachedData) {
+      return {
+        request,
+        response: {
+          id: request.id,
+          jsonrpc: request.jsonrpc || '2.0',
+          result: cachedData.result,
+        },
+        shouldForward: false,
+        rateLimited: false,
+        fromCache: true,
+      }
+    }
+
+    // If not in cache, mark for later caching
+    return {
+      request,
+      response: null,
+      shouldForward: true,
+      rateLimited: false,
+      shouldCacheENS: true,
```

Comment:
I don't like that this is specifuc to ENS.

Reply:
It should just be `shouldCache` so we can easily add more caching later if needed